spec.html

```diff
@@ -90,13 +91,108 @@ <h1>JSON.parse ( _text_ [ , _reviver_ ] )</h1>
+              1. Assert: _typedValNode_ is an |ArrayLiteral| Parse Node.
+              1. Let _contentNodes_ be ArrayLiteralContentNodes of _typedValNode_.
+              1. Let _len_ be the number of elements in _contentNodes_.
+              1. Let _valLen_ be ! LengthOfArrayLike(_val_).
+              1. Assert: _valLen_ = _len_.
+              1. Let _I_ be 0.
+              1. Repeat, while _I_ &lt; _len_,
+                1. Let _propName_ be ! ToString(𝔽(_I_)).
+                1. Let _elementParseRecord_ be CreateJSONParseRecord(_contentNodes_[_I_], _propName_, ! Get(_val_, _propName_)).
+                1. Append _elementParseRecord_ to _elements_.
+                1. Set _I_ to _I_ + 1.
+            1. Else,
+              1. Assert: _typedValNode_ is an |ObjectLiteral| Parse Node.
+              1. Let _properties_ be PropertyDefinitionList of _typedValNode_.
+              1. NOTE: Because _val_ was produced from JSON text and has not been modified, all of its property keys are Strings and will be exhaustively enumerated in source text order.
+              1. Let _keys_ be ! EnumerableOwnProperties(_val_, ~key~).
+              1. For each String _P_ of _keys_, do
+                1. Let _propertyDefinition_ be ~empty~.
+                1. For each Parse Node _candidate_ of _properties_, do
+                  1. Let _propName_ be PropName of _candidate_.
+                  1. If SameValue(_propName_, _P_) is *true*, set _propertyDefinition_ to _candidate_.
+                1. Assert: _propertyDefinition_ is not ~empty~.
+                1. NOTE: In the case of JSON text specifying multiple name/value pairs with the same name for a single object (such as <code>{"a":"lost","a":"kept"}</code>), the value for the corresponding property of the resulting ECMAScript object is specified by the last pair with that name.
+                1. Assert: _propertyDefinition_ is <emu-grammar>PropertyDefinition : PropertyName `:` AssignmentExpression</emu-grammar>.
+                1. Let _propertyValueNode_ be the |AssignmentExpression| of _propertyDefinition_.
+                1. Let _entryParseRecord_ be CreateJSONParseRecord(_propertyValueNode_, _P_, ! Get(_val_, _P_)).
+                1. Append _entryParseRecord_ to _entries_.
+          1. Else,
+            1. Assert: _typedValNode_ is not an |ArrayLiteral| Parse Node and not an |ObjectLiteral| Parse Node.
+          1. Return the JSON Parse Record { [[ParseNode]]: _typedValNode_, [[Key]]: _key_, [[Value]]: _val_, [[Elements]]: _elements_, [[Entries]]: _entries_ }.
+        </emu-alg>
+      </emu-clause>
+    </ins>
+
     <emu-clause id="sec-internalizejsonproperty" type="abstract operation">
       <h1>
         InternalizeJSONProperty (
           _holder_: an Object,
           _name_: a String,
           _reviver_: a function object,
-          <ins>_valNode_: a Parse Node,</ins>
+          <ins>_parseRecord_: either a JSON Parse Record or *undefined*,</ins>
```

Comment:
```suggestion
          <ins>_parseRecord_: either a JSON Parse Record or ~empty~,</ins>
```

And mutadis mutandis below.

Reply:
Sure, done.